### PR TITLE
[Serve] Add default app name

### PR DESCRIPTION
**Pull request by sihanwang41**
## Why are these changes needed?

- Add default name for applications deploy.
- Fix a bug for not setting the deployment name correctly for function deployment.
- Fix unit tests because of the default application name change.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ollie-iterators** [comment]
The documentation test should be fixed by merging in recent changes

File: python/ray/serve/_private/deployment_graph_build.py [hunk 1]
```diff
@@ -264,6 +265,14 @@ def replace_with_handle(node):
                 dag_node._body.__annotations__["return"]
             )
 
+        if "deployment_schema" in dag_node._bound_other_args_to_resolve:
```
Comment (edoakes) [inline python/ray/serve/_private/deployment_graph_build.py:268]: add a comment this block is unintelligible lol

File: python/ray/serve/_private/constants.py [hunk 2]
```diff
@@ -29,7 +29,7 @@
 DEFAULT_GRPC_PORT = 9000
 
 #: Default Serve application name
-SERVE_DEFAULT_APP_NAME = ""
+SERVE_DEFAULT_APP_NAME = "DEFAULT_RAY_SERVE_APP"
```
Comment (edoakes) [inline python/ray/serve/_private/constants.py:32]: this is potentially user facing so this name is a bit ugly, should we just make it "default" ?

**Review by edoakes — APPROVED** [review]

**Comment by sihanwang41** [comment]
test failure is not related to serve.

**Comment by kevin85421** [comment]
Hi, this PR seems to cause the compatibility test in KubeRay failing consistently. See https://github.com/ray-project/ray/issues/34799 for more details. Thanks!

**Pull request merged**